next/lib/graphql/queries.ts

```diff
@@ -54,6 +54,48 @@ export const GET_STATIC_PATHS = graphql(`
   }
 `);
 
+/**
+ * This query is used to get all nodes for the sitemap.
+ * When a new node is created, it should be added here.
+ */
+export const GET_SITEMAP_NODES = graphql(`
+  query getSitemapNodes($page: Int = 0, $langcode: String = "en") {
+    sitemapNodes(filter: { langcode: $langcode }, page: $page) {
+      results {
+        ... on NodeInterface {
+          path
+          changed {
+            timestamp
+          }
+          langcode {
+            id
+          }
+        }
+        ... on NodePage {
+          translations {
+            ...FragmentNodeTranslation
+          }
+        }
+        ... on NodeArticle {
+          translations {
+            ...FragmentNodeTranslation
+          }
+        }
+        ... on NodeFrontpage {
+          translations {
+            ...FragmentNodeTranslation
+          }
+        }
```

Comment:
the NodeInterface does not include the translations field (you might leave some content types untraslated) so we have to add the translations field to each node type here.